### PR TITLE
*: do not unify read pool by default if legacy pools are configured

### PR DESCRIPTION
**Pull request by sticnarf**
###  What have you changed?

This PR changes the way TiKV deals with read pool configurations.

The current TiKV will unify read pools and use the configurations in the `[readpool.unified]` section. However, it may lead to unexpected thread number if the user has configured the legacy pools and just upgraded to the new version.

So we change the behavior here. If `readpool.unify-read-pool` is not specified and the storage or coprocessor readpool configurations don't remain default, we won't unify the read pools.

###  What is the type of the changes?

- Misc (other changes)

###  How is the PR tested?

- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

Should be mentioned in the release note.

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -58,7 +58,8 @@
 ## Max running tasks of each worker, reject if exceeded.
 # max-tasks-per-worker = 2000
 
-[readpool.storage]
+# Uncomment this section if you don't intend to unify the read pools.
```
Comment (BusyJay) [inline etc/config-template.toml]: The comment is confusing. What if I keep both this section and unify thread pool section?

File: src/config.rs [hunk 2]
```diff
@@ -1658,14 +1658,47 @@ impl ReadPoolConfig {
...
 }
 
-impl Default for ReadPoolConfig {
-    fn default() -> ReadPoolConfig {
-        ReadPoolConfig {
-            unify_read_pool: true,
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+struct RawReadPoolConfig {
```
Comment (BusyJay) [inline src/config.rs]: Why not change in place?

File: etc/config-template.toml (hunk 1 above)
Comment (sticnarf) [reply etc/config-template.toml]: I don't have a good idea about how to know whether the user has configured the old thread pools.

**Comment by sticnarf** [comment]
@BusyJay Now I choose to compare the configuration with default to determine whether the user has configured manually.

File: tests/integrations/config/test-default.toml [hunk 3]
```diff
@@ -1,7 +1,11 @@
-[readpool.coprocessor]
+[readpool]
+
+[readpool.unified]
```
Comment (BusyJay) [inline tests/integrations/config/test-default.toml]: Why the file is changed?

Comment (sticnarf) [reply tests/integrations/config/test-default.toml]: Update it to be identical to the current config template.

**Comment by sticnarf** [comment]
> Any tests to cover?

Test added

File: tests/integrations/config/test-default.toml (hunk 3 above)
Comment (BusyJay) [reply tests/integrations/config/test-default.toml]: But there is no new configuration entry, why the output adds new `readpool` and `readpool.unified` tables?

Comment (sticnarf) [reply tests/integrations/config/test-default.toml]: This file is only read and compared with `TiKvConfig::default()`. We don't deserialize the config and compare with this file.
That is, this change is just a late addition. I think this file should be synchronized with `etc/config-template.toml`.

**Pull request merged**